packages/nc-gui/components/cell/attachment/Carousel.vue

```diff
@@ -1,156 +1,361 @@
 
-const imageItems = computed(() => visibleItems.value.filter((item) => isImage(item.title, item.mimetype)))
+/*
+const openComments = ref(false)
+*/
+
+const { isUIAllowed } = useRoles()
+
+const container = ref<HTMLElement | null>(null)
+
+const emblaMainApi: CarouselApi = ref()
+const emblaThumbnailApi: CarouselApi = ref()
+const selectedIndex = ref()
+
+const filetoDelete = reactive({
+  title: '',
+  i: 0,
+})
+const isModalOpen = ref(false)
+
+function onRemoveFileClick(title: any, i: number) {
+  isModalOpen.value = true
+  filetoDelete.title = title
+  filetoDelete.i = i
+}
+
+const handleFileDelete = (i: number) => {
+  removeFile(i)
+  isModalOpen.value = false
+  filetoDelete.i = 0
+  filetoDelete.title = ''
+}
 
 const { getPossibleAttachmentSrc } = useAttachment()
 
-/** navigate to previous image on button click */
-onKeyDown(
-  (e) => ['Left', 'ArrowLeft', 'A'].includes(e.key),
-  () => {
-    if (carouselRef.value) carouselRef.value.prev()
-  },
-)
-
-/** navigate to next image on button click */
-onKeyDown(
-  (e) => ['Right', 'ArrowRight', 'D'].includes(e.key),
-  () => {
-    if (carouselRef.value) carouselRef.value.next()
-  },
-)
-
-/** set our selected image when slide changes */
-function onSlideChange(index: number) {
-  selectedImage.value = imageItems.value[index]
+useEventListener(container, 'click', (e) => {
+  const target = e.target as HTMLElement
+  if (!target.closest('.keep-open') && !target.closest('.nc-button') && !target?.closest('img') && !target?.closest('video')) {
+    selectedFile.value = false
+  }
+})
+
+const onThumbClick = (index: number) => {
+  if (!emblaMainApi.value || !emblaThumbnailApi.value) return
+
+  emblaMainApi.value.scrollTo(index)
+  emblaThumbnailApi.value.scrollTo(index)
+}
+
+const onSelect = () => {
+  if (!emblaMainApi.value || !emblaThumbnailApi.value) return
+
+  const newSnap = emblaMainApi.value.selectedScrollSnap()
+
+  selectedIndex.value = newSnap
+  selectedFile.value = visibleItems.value[newSnap]
+  emblaThumbnailApi.value.scrollTo(newSnap)
 }
 
-/** set our carousel ref and move to initial slide  */
-const setCarouselRef = (el: Element) => {
-  carouselRef.value = el
+const goPrev = () => {
+  if (!emblaMainApi.value || !emblaThumbnailApi.value) return
 
-  carouselRef.value?.goTo(
-    imageItems.value.findIndex((item) => item === selectedImage.value),
-    true,
-  )
+  emblaMainApi.value.scrollPrev()
+  emblaThumbnailApi.value.scrollPrev()
 }
 
-/** close overlay view when clicking outside of image */
-useEventListener(container, 'click', (e) => {
-  if (!(e.target as HTMLElement)?.closest('.keep-open') && !(e.target as HTMLElement)?.closest('img')) {
-    selectedImage.value = false
+const goNext = () => {
+  if (!emblaMainApi.value || !emblaThumbnailApi.value) return
+
+  emblaMainApi.value.scrollNext()
+  emblaThumbnailApi.value.scrollNext()
+}
+
+watchOnce(emblaMainApi, async (emblaMainApi) => {
+  if (!emblaMainApi) return
+
+  container.value?.focus()
+
```

Comment:
```
When the carousel is initialized, we set the selected index to the index of the selected file
and scroll to that index. We only need to do this once, so we use watchOnce.
The focus is set to the container so that the keyboard navigation works
```

Reply:
initially it could be null as well, right?

Reply:
you meant the emblaMainApi?